### PR TITLE
WT-9960 Updating CppSuite bounded_cursor_stress test to better handle WT_ROLLBACK

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: test/cppsuite/tests/bounded_cursor_stress.cpp [hunk 1]
```diff
@@ -681,6 +564,141 @@ class bounded_cursor_stress : public test {
...
+                range_ret = bounded_cursor->prev(bounded_cursor.get());
+            }
+
+            if (normal_ret != 0 && normal_ret != WT_NOTFOUND)
+                return normal_ret;
+
+            if (range_ret != 0 && range_ret != WT_NOTFOUND)
+                return range_ret;
+
+            testutil_assert(normal_ret == 0 || normal_ret == WT_NOTFOUND);
```
Comment (clarissecheah) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:666]: Do we still need these asserts if we have the if blocks from l660-664?

**Review by clarissecheah — COMMENTED** [review]
Just left a couple questions

File: test/cppsuite/tests/bounded_cursor_stress.cpp [hunk 2]
```diff
@@ -681,6 +564,141 @@ class bounded_cursor_stress : public test {
...
+
+        /* Retrieve the key the cursor is pointing at. */
+        const char *range_key;
+        testutil_check(normal_cursor->get_key(normal_cursor.get(), &normal_key));
+        testutil_check(bounded_cursor->get_key(bounded_cursor.get(), &range_key));
+        testutil_assert(std::string(normal_key).compare(range_key) == 0);
+
+        return cursor_traversal_walk(bounded_cursor, normal_cursor, lower_bound, upper_bound, next);
+    }
+
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply test/cppsuite/tests/bounded_cursor_stress.cpp:644]: 015d1db80

File: test/cppsuite/tests/bounded_cursor_stress.cpp (hunk 1 above)
Comment (luke-pearson) [reply test/cppsuite/tests/bounded_cursor_stress.cpp]: 015d1db80

File: test/cppsuite/tests/bounded_cursor_stress.cpp [hunk 3]
```diff
@@ -681,6 +564,141 @@ class bounded_cursor_stress : public test {
...
+                return normal_ret;
+
+            if (range_ret != 0 && range_ret != WT_NOTFOUND)
+                return range_ret;
+
+            char *normal_key, *range_key;
+            const std::string &lower_key = lower_bound.get_key();
+            const std::string &upper_key = upper_bound.get_key();
+            /* Early exit if we have reached the end of the table. */
+            if (range_ret == WT_NOTFOUND && normal_ret == WT_NOTFOUND)
```
Comment (jiechenbo) [inline test/cppsuite/tests/bounded_cursor_stress.cpp:673]: we should add an assert to check that it is not possible to have `range_ret == WT_NOTFOUND && normal_ret = 0` I should have added this check earlier my mistake

**Review by jiechenbo — APPROVED** [review]
LGTM! Just one comment

File: test/cppsuite/tests/bounded_cursor_stress.cpp (hunk 3 above)
Comment (luke-pearson) [reply test/cppsuite/tests/bounded_cursor_stress.cpp:673]: That case is checked on line 676 am I missing something?

**Review by clarissecheah — APPROVED** [review]
LGTM

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Pull request merged**